OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -605,8 +605,10 @@ public void ResolveOrder(Actor self, Order order)
 				{
 					if (IsPlane)
 						self.QueueActivity(new ReturnToBase(self, Info.AbortOnResupply));
-					else
+					else if (!Info.RepairBuildings.Contains(order.TargetActor.Info.Name))
```

Comment:
I agree to do it for both Heli and Plane, but IMO the suggested change is worse. Currently in Activity.ResolveOrder you have various logic. This Repair check can be done once for both Plane/Heli in one place instead of copying it to 2 classes (to be rewritten) which currently are at least consistent that they always return to Base as their name also suggest. It is more logical not to call ReturnToBase activities when we know we don't want it.